packages/beacon-node/src/network/network.ts

```diff
@@ -224,6 +222,9 @@ export class Network implements INetwork {
     this.chain.emitter.off(routes.events.EventType.lightClientOptimisticUpdate, this.onLightClientOptimisticUpdate);
     await this.core.close();
     this.logger.debug("network core closed");
+
+    // Used only for sleep() statements
+    this.controller.abort();
```

Comment:
This change protects from scenarios where connection and libp2p dependencies are cleanup before properly saying goodbye. Identified through a lot of errors during e2e tests. 

```
May-02 11:06:48.905[reqresp-main-B] verbose: Req  error method=goodbye, version=1, encoding=ssz_snappy, client=Lodestar, peer=16...koSwxP, requestId=6 - REQUEST_ERROR_DIAL_ERROR
```

Reply:
so this is similar / same rational as I noted here https://github.com/ChainSafe/lodestar/pull/5330#discussion_r1202816552?

> In general, it is best to abort as late as possible in shutdown sequence. This ensures all intervals/timeouts are cleared first and then all remaining pending operations are aborted. Prevents edge case where abort is done and interval schedules another operation which would be stuck pending and delay shutdown.

or is this different?

Reply:
Yes it's the same rationale. And yes it was the same problem that earlier controller abort was clearing up some deep event handler and later shutdown of the netowrk core was causing this failure to send `goodbye` message.